Source/WebCore/css/StyleRule.cpp

```diff
@@ -199,30 +199,48 @@ Ref<CSSRule> StyleRuleBase::createCSSOMWrapper(CSSStyleSheet* parentSheet, CSSGr
 
 unsigned StyleRule::averageSizeInBytes()
 {
-    return sizeof(StyleRule) + sizeof(CSSSelector) + StyleProperties::averageSizeInBytes();
+    return sizeof(StyleRule) + sizeof(CSSSelector) + StyleProperties::averageSizeInBytes() + sizeof(WeakPtr<StyleRule>) + sizeof(Vector<Ref<StyleRuleBase>>);
 }
 
-StyleRule::StyleRule(Ref<StyleProperties>&& properties, bool hasDocumentSecurityOrigin, CSSSelectorList&& selectors)
+StyleRule::StyleRule(Ref<StyleProperties>&& properties, bool hasDocumentSecurityOrigin, CSSSelectorList&& selectors, WeakPtr<StyleRule> parentRule, Vector<Ref<StyleRule>> nestedRules)
```

Comment:
Could we make `Vector<Ref<StyleRule>> nestedRules` a move operation instead of a copy? Or pass as a const reference so we only copy on assignment to the member?